### PR TITLE
TST: added IGRF unit tests

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Addresses #5 by adding some unit tests and enabling this testing in the CI.

# Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Ran unit tests locally.

`python -m pytest`

**Test Configuration**:
* Operating system: OS X Big Sur
* Version number: Python 3.9
* Any details about your local setup that are relevant

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update zenodo.json file for new code contributors

### CODE REVIEW

**Comment by aburrell** [comment]
@jklenzing could you look at how I set up coveralls maybe see why some of the tests are failing?  I tried fixing it, but without success 🐛

**Review by jklenzing — COMMENTED** [review]
I don't see anything that leaps out at met as wrong compared to how we've set up the tests elsewhere. My first thought is whether this is on the server side. ~~Have you tried re-running the tests in case coveralls was down temporarily?~~

EDIT: just saw the multiple actions that failed for the pull, but not push. develop is working as expected.

File: pyproject.toml [hunk 1]
```diff
@@ -45,6 +45,9 @@ test = [
   "flake8",
   "flake8-docstrings",
   "hacking >= 1.0",
+  "pytest",
+  "pytest-cov",
+  "pytest-ordering"
```
[the remark this answers is not in the document]
Comment (aburrell) [reply pyproject.toml:50]: It was 🐱 will delete and rerun!

Comment (aburrell) [inline pyproject.toml:50]:
```suggestion
```

**Comment by jklenzing** [comment]
Could be coveralls / package related. Similar issues with the push being fine and the pull mostly failing at https://github.com/pysat/pysatNASA/pull/214

**Comment by aburrell** [comment]
Your new fix worked @jklenzing! 😸

File: PyIRI/tests/test_igrf_lib.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        self.eff_out = np.array([155.88457268, 138.56406461, 121.24355653,
+                                 103.92304845, 86.60254038, 69.28203230,
+                                 51.96152423, 34.64101615, 17.32050808,
+                                 0.00000000, 17.32050808, 34.64101615,
+                                 51.96152423, 69.28203230, 86.60254038,
+                                 103.92304845, 121.24355653, 138.56406461])
+        return
+
+    def teardown_method(self):
+        """Clean up all tests."""
```
Comment (jklenzing) [inline PyIRI/tests/test_igrf_lib.py:63]:
```suggestion
        """Clean up after every test."""
```

**Review by jklenzing — CHANGES_REQUESTED** [review]

**Review by jklenzing — APPROVED** [review]

**Pull request merged**